### PR TITLE
Implement pagination control; update color system

### PR DESCRIPTION
**Pull request by jgolieb**
This PR implements the pagination control per the latest spec discussed in our last team meeting. I also made some changes to the color system, notably to remove the unnecessary `$fluid-` prefixes and to separate text colors from other types of colors.

### CODE REVIEW

File: docs/index.html [hunk 1]
```diff
@@ -776,21 +776,40 @@ <h2>
...
-                    <span aria-hidden="true"><i class="fa fa-angle-right"></i></span>
-                    <span class="sr-only">Next</span>
-                  </a>
-                </li>
-              </ul>
+            <nav class="pagination" aria-label="Page navigation example">
+              <div class="fade">
+                <div class="form-group">
+                  <label for="itemsPerPage">Show:</label>
+                  <select id="itemsPerPage">
```
Comment (tobias-istvan) [inline docs/index.html:783]: I think we should use dropdown here as it is on the mock.

File: docs/index.html [hunk 2]
```diff
@@ -776,21 +776,40 @@ <h2>
...
+                    <li class="page-item"><a class="page-link" href="#">2</a></li>
+                    <li class="page-item"><a class="page-link" href="#">3</a></li>
+                    <li class="page-item disabled"><a class="page-link" href="#">&hellip;</a></li>
+                    <li class="page-item"><a class="page-link" href="#">500</a></li>
+                    <li class="page-item disabled"><a class="page-link" href="#">&hellip;</a></li>
+                    <li class="page-item"><a class="page-link" href="#">1000</a></li>
+                  </ul>
+                </div>
+              </div>
+              <span class="fade show">Showing 1-10 of 1000</span>
```
Comment (tobias-istvan) [inline docs/index.html]: In the specification there is no label `Showing`.

File: src/components/_pagination.scss [hunk 3]
```diff
@@ -6,6 +6,111 @@
...
+  justify-content: flex-end;
+
+  > span.fade.show {
+    white-space: nowrap;
+    align-self: center;
+    margin-right: px(4);
+  }
+
+  > div.fade {
+    position: absolute;
```
Comment (tobias-istvan) [inline src/components/_pagination.scss:21]: The transition looks good. However I don't like the magic calculation (`right: $input-height * 2;`).
I would prefer setting the `display` prop or playing a bit with animations.

File: docs/index.html (hunk 2 above)
Comment (jgolieb) [reply docs/index.html]: You're right! I don't know where I got that from.

File: docs/index.html (hunk 1 above)
Comment (jgolieb) [reply docs/index.html:783]: Unless I'm missing something, Bootstrap dropdowns are for displaying menus or other hidden content, not for using as input fields. Of course, the downside of using a <select> is that we can't style the popup part of it, but to use a custom dropdown instead would require creating a fully custom control (probably would need JS as well).

File: src/components/_pagination.scss (hunk 3 above)
Comment (jgolieb) [reply src/components/_pagination.scss:21]: That calculation has nothing to do with the animation. It just shifts the absolutely positioned "Show: [] Go: []" controls to the left past the Prev/Next arrows. Each of those arrows is $input-height wide, so multiplying by 2 gives us the correct amount to shift over. So it's not magic.

File: docs/index.html (hunk 1 above)
Comment (tobias-istvan) [reply docs/index.html:783]: That is correct. However on the mockup we have dropdown. We need to discuss if we will create a new component for this use case.

File: src/components/_pagination.scss (hunk 3 above)
Comment (tobias-istvan) [reply src/components/_pagination.scss:21]: I mean I would add animation with keyframes if we want some effect on the fade in and out.
However I would prefer the more simple positioning with no effect. 
I said it is magic because the name is misleading and also brings some limitation/extra effort for custom pagination.

File: docs/index.html (hunk 1 above)
Comment (jgolieb) [reply docs/index.html:783]: The "mockup" is just a screenshot from the DAS implementation. We can confirm with Houshang whether the <select> usage is ok. I really don't think he intended for us to create a new component just for that.

File: src/components/_pagination.scss (hunk 3 above)
Comment (jgolieb) [reply src/components/_pagination.scss:21]: I've updated this comment because I re-read what you said and I think I understand it a little better now, so let me try again. :)

- The fade effect using the .fade class is already part of Bootstrap, and certainly a lot simpler than creating a keyframe animation. Not sure why you object to using the fade effect.

- If you just don't like the name of the variable, then ok I can easily create a more appropriately named one. I don't know what you mean about "limitation/extra effort".

File: src/components/_navbar.scss [hunk 4]
```diff
@@ -47,7 +47,7 @@
 
     .navbar-nav {
 
-      .nav-item.nav-link, 
+      .nav-item.nav-link,
       .nav-item > .nav-link {
         height: px(38) !important;
```
Comment (mihai-mihalache) [inline src/components/_navbar.scss]: I would avoid using !important when possible. Could you maybe increase the specificity of the selector instead?

File: src/components/_navbar.scss [hunk 5]
```diff
@@ -312,16 +312,16 @@
...
-  background-color: $fluid-navbar-brand-bg;
+  color: $navbar-brand-color !important;
+  background-color: $navbar-brand-bg;
   text-transform: uppercase;
   white-space: nowrap;
   line-height: px(60);
   overflow: hidden;
   @include transition($transition-collapse-all);
 
   &[data-toggle="collapse"] {
```
Comment (mihai-mihalache) [inline src/components/_navbar.scss]: Tight coupling this theme to a specific JS framework is bad practice.

Comment (jgolieb) [reply src/components/_navbar.scss]: I assume you're referring again to the data-toggle="collapse" thing. Again, this is not relevant to this PR. However... we're explicitly tight-coupling this to Bootstrap and this is part of Bootstrap. The other way would be to add a custom class and that would break with our design goals.

File: src/components/_navbar.scss (hunk 4 above)
Comment (jgolieb) [reply src/components/_navbar.scss]: Even though I agree with where you're coming from here, you're commenting on something that is outside the scope of this PR. The only change on that line is whitespace.

File: src/components/_navbar.scss (hunk 5 above)
Comment (mihai-mihalache) [reply src/components/_navbar.scss]: Yes, I am referring to the **data-toggle="collapse"** thing and no, you're not tight coupling to Bootstrap, you're tight coupling to jQuery which is not in the specs for Fluid. I can also say with 100% certainty  that there is nothing in Bootstrap with a "data-\*" type selector (I've searched). Thus, either by adding a class (preferred) or by adding a "data-\*" attribute selector you should explicitly state this in the documentation for this component.

Comment (mihai-mihalache) [reply src/components/_navbar.scss]: Also sorry for referencing old PR code but I only now have noticed this :disappointed:

Comment (jgolieb) [reply src/components/_navbar.scss]: But 1) Bootstrap is tight-coupled to jQuery and 2) look at the docs on Collapse here: https://getbootstrap.com/docs/4.1/components/collapse/ -- it uses data-toggle="collapse", even explicitly stating that "...the data-toggle="collapse" is required."

Comment (jgolieb) [reply src/components/_navbar.scss]: And again, if this is your only problem with this PR, then please let's discuss it under a separate issue and not hold up this PR. I have a ton of work waiting after this.

**Review by mihai-mihalache — APPROVED** [review]
